push_swap/src/utils/validate.c

```diff
@@ -0,0 +1,98 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   validate.c                                         :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: jkhasiza <jkhasiza@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/01/05 15:19:00 by codespace         #+#    #+#             */
+/*   Updated: 2024/01/16 18:22:40 by jkhasiza         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "../../include/push_swap.h"
+
+t_bool	validate_4_digits(char **raw_numbers, int start)
+{
+	int	col;
+	int	row;
+
+	row = start;
+	while (raw_numbers[++row])
+	{
+		col = -1;
+		if (raw_numbers[row][0] == '-' || raw_numbers[row][0] == '+')
+			col++;
+		while (ft_isint(raw_numbers[row]) == FALSE)
+			return (FALSE);
+	}
+	return (TRUE);
+}
+
+t_bool	validate_stack_4_duplicates(t_number **stack)
+{
+	t_number	*tmp;
+
+	if (!stack || !(*stack))
+		return (FALSE);
+	tmp = *stack;
+	while (tmp)
+	{
+		if (tmp->next && ft_stackindex(&tmp->next, tmp->number) != -1)
+			return (FALSE);
+		tmp = tmp->next;
+	}
+	return (TRUE);
+}
+
+t_bool	validate_4_overflow(char **raw_numbers, int start)
+{
+	int		row;
+	char	*curr;
+	t_bool	has_sign;
+
+	row = start;
+	while (raw_numbers[++row])
+	{
+		curr = raw_numbers[row];
+		has_sign = FALSE;
+		if (curr[0] == '-' || curr[0] == '+')
+			has_sign = TRUE;
+		if ((!has_sign && ft_strlen(raw_numbers[row]) > 10)
+			|| (has_sign && ft_strlen(raw_numbers[row]) > 11))
+			return (FALSE);
+		if (ft_atoi_lli(curr) > 2147483647 || ft_atoi_lli(curr) < -2147483648)
+			return (FALSE);
```

Comment:
This logic considers `0000000000000000005` as invalid which is wrong and it should have been fine. Fix it.